src/components/DatePicker.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { View } from 'react-native';
+import { Button, List, useTheme } from 'react-native-paper';
+import { DatePickerModal } from 'react-native-paper-dates';
+
+import { useTranslation } from 'language/LanguageProvider';
+import Storage from 'storage';
+
+
+/**
+ * A date picker component that is opened with a press of a button.
+ *
+ * @example
+ * ```
+ * const [date, setDate] = React.useState(new Date());
+ * 
+ * return (
+ *   <DatePicker date={date} setDate={setDate} />
+ * );
+ * 
+ * ```
+ * 
+ * @param date - End time value as a Date object (eg a hook)
+ * @param setDate -End time set function (eg a hook)
+ * 
+ * @param style - Optional styling of the button
+ * @param containerStyle - Optional styling of the view container
+ * 
+ * @returns The TimePicker component
+ *
+ */
+export const DatePicker = (props: {date: Date, setDate: React.Dispatch<React.SetStateAction<Date>>, style?: {}, containerStyle?: {}}) => {
+  const lang = useTranslation();
+  const [settings, modifySettings] = Storage.useSettings();
+  const [open, setOpen] = React.useState(false);
+
+  const onDismissSingle = React.useCallback(() => {
+    setOpen(false);
+  }, [setOpen]);
+
+  const onConfirmSingle = React.useCallback(
+    (params) => {
+      setOpen(false);
+      props.setDate(params.date);
+    },
+    [setOpen, props.setDate]
+  );
```

Comment:
I am quite unsure on when to use useCallback, but it seems according to this article that it is not worth to wrap simple arrow functions with it?


"Because inline functions are cheap, the re-creation of functions on each rendering is not a problem. A few inline functions per component are acceptable."

https://dmitripavlutin.com/dont-overuse-react-usecallback/

Reply:
In this case i agree that callbacks are a bit overkill. It mainly came from an example and seemed like a good thing to use, but since it bring more overhead than what it prevents it's not really necessary here.